iothub/device/src/IotHubModuleClient.cs

```diff
@@ -280,6 +280,7 @@ private async Task<DirectMethodResponse> InvokeMethodAsync(Uri uri, DirectMethod
             finally
             {
                 httpClientHandler?.Dispose();
+                _certValidator.Dispose();
```

Comment:
Does this need NRE protection?

Reply:
It does not. We create an instance NullCertificateValidator in such case.